### PR TITLE
Clean up workspace compilation and testing.

### LINKED ISSUE #192: CI does not test 'png' feature
(I am assuming, because some breakage gated on this feature presumably passed CI. see #191)

### PR DESCRIPTION
**Pull request by xStrom**
This PR refactors the CI script to allow for easier expansion and testing of features. The cargo manifest files are cleaned up a bit and unified in style. I also updated `README.md` to better reflect the current state of things.

Fixes  #192.

### CODE REVIEW

File: piet-common/Cargo.toml [hunk 1]
```diff
@@ -14,34 +14,36 @@ cairo = ["piet-cairo", "cairo-rs"]
...
-cairo-rs = { version = "0.8.1", default_features = false, optional = true}
 
-[target.'cfg(not(any(target_arch="wasm32", target_os="windows", target_os="macos")))'.dependencies]
+cfg-if = "0.1.10"
+cairo-rs = { version = "0.8.1", default_features = false, optional = true }
+png = { version = "0.16.1", optional = true }
+
+[target.'cfg(target_os="linux")'.dependencies]
 piet-cairo = { version = "0.1.0", path = "../piet-cairo" }
 cairo-rs = { version = "0.8.1", default_features = false}
```
Comment (cmyr) [inline piet-common/Cargo.toml:29]: will these get added to a wasm32 build on linux? or is that not a thing?

File: piet-common/Cargo.toml [hunk 2]
```diff
@@ -14,34 +14,36 @@ cairo = ["piet-cairo", "cairo-rs"]
...
 piet-web = { version = "0.1.0", path = "../piet-web", optional = true }
-cairo-rs = { version = "0.8.1", default_features = false, optional = true}
 
-[target.'cfg(not(any(target_arch="wasm32", target_os="windows", target_os="macos")))'.dependencies]
+cfg-if = "0.1.10"
+cairo-rs = { version = "0.8.1", default_features = false, optional = true }
+png = { version = "0.16.1", optional = true }
+
+[target.'cfg(target_os="linux")'.dependencies]
 piet-cairo = { version = "0.1.0", path = "../piet-cairo" }
```
Comment (cmyr) [inline piet-common/Cargo.toml:28]: we want piet-cairo to be built if the --features=cairo option is passed on macOS. Maybe we're being too fancy though, and could just get rid of this option?

File: piet-common/src/cairo_back.rs [hunk 3]
```diff
@@ -136,14 +136,14 @@ impl<'a> BitmapTarget<'a> {
...
         let width = self.surface.get_width();
         let image = self.into_raw_pixels(ImageFormat::RgbaPremul)?;
-        let file = BufWriter::new(File::create(path).map_err(|e| Into::<Box<_>>::into(e))?);
+        let file = BufWriter::new(File::create(path).map_err(Into::<Box<_>>::into)?);
         let mut encoder = Encoder::new(file, width as u32, height as u32);
         encoder.set_color(ColorType::RGBA);
         encoder
             .write_header()
-            .map_err(|e| Into::<Box<_>>::into(e))?
+            .map_err(Into::<Box<_>>::into)?
```
Comment (cmyr) [inline piet-common/src/cairo_back.rs:144]: huh I tried to do this recently and couldn't get it to work

File: piet-common/Cargo.toml (hunk 1 above)
Comment (xStrom) [reply piet-common/Cargo.toml:29]: They will not be compiled. There are no warnings or anything either, so I think it's fine to not make the cfg guard more complex here.

File: piet-common/Cargo.toml (hunk 2 above)
Comment (xStrom) [reply piet-common/Cargo.toml:28]: Yes the `cairo` feature still works on every platform, including Windows when using gcc.

The CI only tests cairo on Linux. If we want to ensure that cairo support is awesome then we could test it on macOS and Windows too. However my understanding has been that cairo is more of a temporary backend, so probably fine to just test it on Linux.

**Pull request merged**